File: Array/linchuanxin/Algorithm/day45_0715_移动零/day45_leetcode283.java
package day45_0715;

public class day45_leetcode283 {

    public static void main(String[] args) {
        int[] arr = {0,1,0,3,12};
        moveZeroes(arr);
    }

    public static void moveZeroes(int[] nums) {
        if(nums.length<=1)
            return;
        int j = 0;
        for (int i = 0; i < nums.length; i++) {
                if(nums[i] != 0 ){
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j++] = temp;
                }

        }
        for (int i = 0; i < nums.length; i++) {
            System.out.println(nums[i]);
        }
    }
}

File: Array/linchuanxin/Algorithm/day45_0715_移动零/day45_leetcode283.kt
package day45_0715

fun moveZeroes(nums: IntArray): Unit {

    if(nums == null)
        return
    var j:Int = 0
    //for (i in 0..nums.size-1){
    for (i in 0 until nums.size){
        if(nums[i] != 0){
            var temp:Int = nums[i]
            nums[i] = nums[j]
            nums[j++] = temp
        }
    }

}